src/models/user/stream.rs

```diff
@@ -187,10 +213,55 @@ impl UserStream {
         .await
     }
 
+    async fn get_post_replies_ids(
+        post_id: Option<String>,
+        author_id: Option<String>,
+    ) -> Result<Option<Vec<String>>, DynError> {
+        let post_id = post_id
+            .ok_or("Post ID should be provided for user streams with source 'post_replies'")?;
+        let author_id = author_id
+            .ok_or("Author ID should be provided for user streams with source 'post_replies'")?;
+        let key_parts = [
+            &POST_REPLIES_PER_POST_KEY_PARTS[..],
+            &[author_id.as_str(), post_id.as_str()],
+        ]
+        .concat();
+        let replies = PostStream::try_from_index_sorted_set(
+            &key_parts,
+            None,
+            None,
+            None,
+            None,
+            SortOrder::Descending,
+            None,
+        )
+        .await?;
+        let unique_user_ids: HashSet<String> = replies
+            .map(|replies| {
+                replies
+                    .into_iter()
+                    .map(|reply| {
+                        reply
+                            .0
+                            .split(":")
+                            .map(ToString::to_string)
+                            .collect::<Vec<String>>()[0]
+                            .clone()
+                    })
+                    .collect::<Vec<String>>()
+            })
+            .into_iter()
+            .flatten()
+            .collect();
+        Ok(Some(unique_user_ids.into_iter().collect()))
```

Comment:
Maybe something like this is clearer and panic safe.
```suggestion
        // If there are replies, extract unique user IDs using a HashSet.
        let unique_user_ids: HashSet<String> = if let Some(replies) = replies {
            replies
                .into_iter()
                .filter_map(|reply| reply.0.split(':').next().map(|s| s.to_string()))
                .collect()
        } else {
            // If no replies are found, return None.
            return Ok(None);
        };

        // Convert the HashSet to a Vec. (Note: the ordering will be arbitrary.)
        Ok(Some(unique_user_ids.into_iter().collect()))
```